R/linting.R

```diff
@@ -0,0 +1,48 @@
+#' Use code linting
+#'
+#' `use_lintr()` Adds support for code linting using the `lintr` package in one of three ways:
+#' local interactive linting only, non-failing (post-build) Travis CI linting, or failing local and Travis CI `testthat` unit test.
+#'
+#' @param type Options are \code{local}, \code{travis} or \code{test}.
+#' @export
+use_lintr <- function(type = c("local", "travis", "test")){
+  type <- match.arg(type)
+
+  if(type == "test"){
+    dir.create("inst", showWarnings = FALSE)
+    .lintr <- "inst/.lintr"
+  } else {
+    .lintr <- ".lintr"
+  }
+
+  sink(.lintr)
+  cat(paste0("linters: with_defaults(\n  ", "line_length_linter(80))\n"))
+  sink()
```

Comment:
We have existing helpers for writing (into) files. See the functions in `write.R`.

Reply:
Thank you. I'll look into all these changes I need to make to resolve these inconsistencies vis a vis `usethis` and `fs` functions.